umap/utils.py

```diff
@@ -109,3 +109,33 @@ def gzip_file(from_path, to_path):
     with open(from_path, 'rb') as f_in:
         with gzip.open(to_path, 'wb') as f_out:
             f_out.writelines(f_in)
+
+def user_details(strategy, details, user=None, *args, **kwargs):
```

Comment:
I'd name it something like `oauth_user_details`, given it's in a generic `utils` file, thus without namespace to explicit what `user_details` are related to.
Also, I'd add a line in the docstring to define it's a pipeline stage to be used with social_auth.